src/Workleap.DomainEventPropagation.Publishing/EventPropagationClient.cs

```diff
@@ -41,22 +41,39 @@ private static DomainEventsHandlerDelegate BuildPipeline(DomainEventsHandlerDele
 
     public Task PublishDomainEventAsync<T>(T domainEvent, CancellationToken cancellationToken)
         where T : IDomainEvent
-        => this.PublishDomainEventsAsync(new[] { domainEvent }, cancellationToken);
+        => this.InternalPublishDomainEventsAsync(new[] { domainEvent }, null, cancellationToken);
 
-    public async Task PublishDomainEventsAsync<T>(IEnumerable<T> domainEvents, CancellationToken cancellationToken)
+    public Task PublishDomainEventAsync<T>(T domainEvent, Action<IDomainEventMetadata> domainEventConfiguration, CancellationToken cancellationToken)
```

Comment:
What do you think of?
```suggestion
    public Task PublishDomainEventAsync<T>(T domainEvent, Action<IDomainEventMetadata> configureMetadata, CancellationToken cancellationToken)
```

Reply:
Also, should the parameter be nullable `Action<IDomainEventMetadata>?`?

Reply:
Well, this is aligned with my other comment 😅

As mentionned I'd rename it to `configureDomainEventMetadata`, unless you really prefer the shorter version.

As for it being nullable, this is an overload, so if you don't intend to provide a specific configuration you can skip the parameter entirely. The alternative would lead to calls looking like `.PublishDomainEventAsync(domainEvent, null, cancellationToken)` which is needlessly verbose, and it would also introduce a breaking change.